### PR TITLE
network: process queued remove requests on stop

### LINKED ISSUE #34837: systemd-networkd fails to drop routes & addresses on stop with `KeepConfiguration=no`
### systemd version the issue has been seen with

~~v255.6~~ v256 onwards, reproduces on v256.4

### Used distribution

NixOS unstable 24.11pre-git (since `a5b2fe73` up to `5633bcf`)

### Linux kernel version used

Both 6.6.54-xanmod1, 6.6.54

### CPU architectures issue was seen on

x86_64

### Component

systemd-networkd

### Expected behaviour you didn't see

`[Network].KeepConfiguration=no` is respected and networkd drops IPv4 addresses when stopped

### Unexpected behaviour you saw

After `SYSTEMD_LOG_LEVEL=debug systemctl import-environment SYSTEMD_LOG_LEVEL && systemctl restart systemd-networkd`, networkd logs show that it is attempting to drop addresses and routes but `ip r` and `ip a` show they are still present. With `networkctl down eth0`, this does not happen.

### Steps to reproduce the problem

Run the `systemd-initrd-networkd.doFlush` NixOS test from a nixpkgs newer than 2024-07-29.

Observe [it fails](https://hydra.nixos.org/build/267693996): ([full logs](https://cache.nixos.org/log/69432jppr8pfpjp3aq0z5bs8ri9d5f84-vm-test-run-systemd-initrd-network-flush.drv))
```
[   12.832587] check-flush-start[352]:     inet 10.0.2.15/24 metric 1024 brd 10.0.2.255 scope global dynamic eth0
[   12.837278] check-flush-start[345]: Network configuration survived switch-root; flushBeforeStage2 failed
```

```
$ networkctl cat 99-ethernet-default-dhcp.network
# /etc/systemd/network/99-ethernet-default-dhcp.network
[Match]
Name=en*
Name=eth*
 
[Network]
DHCP=yes
IPv6PrivacyExtensions=kernel
KeepConfiguration=no # KeepConfiguration is set
```

### Additional program output to the terminal or log subsystem illustrating the issue

`networkctl down eth0` correctly drops address:
https://gist.github.com/ckiee/53c0bcdcedbee8f985563cf3fbb8bea0

`systemctl stop systemd-networkd` tries to drop addresses and routes but fails.
https://gist.github.com/ckiee/b30551342e3d05722916c1e34bf007fc

### PR DESCRIPTION
**Pull request by yuwata**
Fixes a regression caused by 85a6f300c14d75d161cbfdb3eaf5af9594400ecd and its later commits. 
Fixes #34837.

### CODE REVIEW

**Comment by ElvishJerricco** [comment]
I rebased this on 256 and it appears to have resolved the issue.

**Comment by yuwata** [comment]
Hmm, with gcc-13 and mold,
```
/home/runner/work/systemd/systemd/build/systemd-networkd: error while loading shared libraries: cannot apply additional memory protection after relocation: Cannot allocate memory
```

File: src/network/networkd-manager.c [hunk 1]
```diff
@@ -422,30 +422,71 @@ static int manager_connect_rtnl(Manager *m, int fd) {
...
 
-static int signal_terminate_callback(sd_event_source *s, const struct signalfd_siginfo *si, void *userdata) {
-        Manager *m = ASSERT_PTR(userdata);
+static int manager_stop(Manager *manager, ManagerState state) {
+        assert(manager);
+        assert(IN_SET(state, MANAGER_TERMINATING, MANAGER_RESTARTING));
 
-        m->restarting = false;
+        if (manager->state != MANAGER_RUNNING)
+                return 0;
```
Comment (poettering) [inline src/network/networkd-manager.c:461]: shouldnt this fail with EBUSY if you pass TERMINATING but we are in RESTARTING?

Comment (poettering) [reply src/network/networkd-manager.c:461]: or dunno. maybe it should override the target?

Comment (poettering) [reply src/network/networkd-manager.c:461]: or maybe it's actually ok this proposed way, but we should do a warning log that we are not going to change direction now

Comment (poettering) [reply src/network/networkd-manager.c:461]: leaning towards the latter

**Comment by yuwata** [comment]
@poettering Thank you for the review. A log message is added to the suggested place. Upgrading the green label.

**Pull request merged**